File: java/src/leetcode/S1.java
package leetcode;

import java.util.HashMap;

public class S1 {

    public class Solution {
        public int[] twoSum(int[] nums, int target) {
            HashMap<Integer, Integer> map = new HashMap<>();
            for (int i = 0; i < nums.length; i++) {
                Integer pair = map.get(target - nums[i]);
                if (pair != null) {
                    return new int[]{pair, i};
                }
                map.put(nums[i], i);
            }
            return null;
        }
    }
}

File: python/leetcode/S1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i in range(len(nums)):
            if dict.get(nums[i]) is None:
                dict[target - nums[i]] = i
            else:
                return [dict[nums[i]], i]